EngineLayer/ClassicSearch/ClassicSearchEngine.cs

```diff
@@ -17,11 +17,11 @@ public class ClassicSearchEngine : MetaMorpheusEngine
         private readonly List<Protein> Proteins;
         private readonly List<Modification> FixedModifications;
         private readonly List<Modification> VariableModifications;
-        private readonly PeptideSpectralMatch[] PeptideSpectralMatches;
+        private readonly PeptideSpectralMatch[][] PeptideSpectralMatches;
```

Comment:
this seems overly complex... do you really need a 2d array of PSMs for the search? aren't you just categorizing the PSM results by cleavage specificity in the FDR (post-processing?)

Reply:
I agree that this is overly complex. This was something @acesnik brought up, in that the highest scoring peptide from each category (full/semi/nonspecific) should be saved so that they don't have to compete against each other using the MetaMorpheus score. Instead, they each get a q-value from their category, and then the psms between categories "compete" based on q-value, instead of on score. In this way, ambiguity between a full and a semi is resolved based on q-value.